src/chapters/typescript/functions.rst

```diff
@@ -0,0 +1,115 @@
+Functions in TypeScript
+=======================
+
+When creating functions in TypeScript, we have many of the same options as in JavaScript.
+We can make anonymous functions, give it a different number of parameters and so on.
+However, when working with functions in TypeScript, we have to keep *type* in mind.
+
+Declaring Functions
+-------------------
+
+Named Functions
+^^^^^^^^^^^^^^^
+
+Let's take a look at a TypeScript function declaration.
+
+.. sourcecode:: typescript
+
+   function myFunction(a: number): number {
+      return a*2;
+   }
+
+Here you can see that we provided the type of the parameters and the type of the value that is returned.
+If the function doesn't have a return value, then you use ``void`` for its type.
+
+.. note::
+
+   While in many cases, it is possible to determine the return type during compilation, if you know the return type, you should give it. 
+
+Anonymous Functions
+^^^^^^^^^^^^^^^^^^^
+
+For an anonymous function, you still need to provide types for the value returned and the parameters.
+
+.. sourcecode:: typescript
+
+   let myFunction = function(a: number): number { return a*2; };
+
+The above code looks very much like an anonymous function declaration with the types included!
+However, TypeScript has arrow notation and here we use the **fat arrow**, ``=>``.
```

Comment:
can we get away without discussing fat arrows? i know we've talked about this syntax in other areas of the text and managed to avoid it while still getting crucial topics across. In this case, it looks like it might be necessary but i can't tell?

Reply:
I think we can! The anonymous function section is short and sweet, just reminding people to include their types!